src/client.ts

```diff
@@ -178,73 +214,129 @@ export class Client {
             if (sendParams) {
               // Request parameters to magic attributes
-              const queryParams = info.context.request.query
-
+              const queryParams = info.request.query
+              const requestBody = info.request.body
+              const params = { ...queryParams, ...requestBody }
               span.setAttribute(
                 "appsignal.request.parameters",
-                JSON.stringify(queryParams)
+                JSON.stringify(params)
+              )
+            }
+
+            if (sendSessionData) {
+              // Session data to magic attributes
+              span.setAttribute(
+                "appsignal.request.session_data",
+                JSON.stringify(info.request.cookies)
               )
             }
           }
-        }),
-        new MySQLInstrumentation(),
-        new MySQL2Instrumentation(),
-        new PgInstrumentation(),
-        new RedisInstrumentation({
-          dbStatementSerializer: RedisDbStatementSerializer
-        }),
-        new Redis4Instrumentation({
-          dbStatementSerializer: RedisDbStatementSerializer
-        }),
-        new IORedisInstrumentation({
-          dbStatementSerializer: RedisDbStatementSerializer
-        }),
-        new PrismaInstrumentation({
-          middleware: true
-        })
-      ]
+        }
+      },
+      "@opentelemetry/instrumentation-http": {
+        headersToSpanAttributes: {
+          server: { requestHeaders }
+        }
+      },
+      "@opentelemetry/instrumentation-ioredis": {
+        dbStatementSerializer: RedisDbStatementSerializer
+      },
+      "@opentelemetry/instrumentation-koa": {
+        requestHook: function (span: OtelSpan, info) {
+          if (sendParams) {
+            // Request parameters to magic attributes
+            const queryParams = info.context.request.query
+
+            span.setAttribute(
+              "appsignal.request.parameters",
+              JSON.stringify(queryParams)
+            )
+          }
+        }
+      },
+      "@opentelemetry/instrumentation-redis": {
+        dbStatementSerializer: RedisDbStatementSerializer
+      },
+      "@opentelemetry/instrumentation-redis-4": {
+        dbStatementSerializer: RedisDbStatementSerializer
+      },
+      "@prisma/instrumentation": {
+        middleware: true
+      }
+    }
+  }
+
+  private instrumentations(
+    instrumentationsOption: AppsignalInstrumentationOption
+  ): InstrumentationOption[] {
+    const defaultConfig = this.defaultInstrumentationsConfig()
+
+    const defaultInstrumentations = Object.entries(DefaultInstrumentationMap)
+      .filter(([key, _instrumentationConstructor]) => {
+        const instrumentationName =
+          key as keyof typeof DefaultInstrumentationMap
+
+        const userInstrumentationConfig =
+          instrumentationsOption[instrumentationName] || {}
+
+        if (userInstrumentationConfig instanceof InstrumentationBase) {
+          return false
```

Comment:
This means it gets filtered out if it's an instance of the instrumentation class?
Let's log a warning that it's being ignored in that case.

It's a bit weird because the additional instrumentation needs to be an instance of the instrumentation, but I understand it's not possible to update the instance with our default config afterwards.